src/lib/dns.ts

```diff
@@ -0,0 +1,19 @@
+import dns from 'node:dns';
+import validator from 'validator';
+import isIpPrivate from 'private-ip';
+
+export const getDnsServers = (getServers: () => string[] = dns.getServers): string[] => {
+	const servers = getServers();
+
+	return servers
+	// Filter out ipv6
+		.filter((addr: string) => {
+			const ipv6Match = /^\[(.*)]/g.exec(addr); // Nested with port
+			return !(validator.isIP(addr, 6) || validator.isIP(ipv6Match?.[1] ?? ''));
+		})
+	// Hide private ips
+		.map((addr: string) => {
+			const ip = addr.replace(/:\d{1,5}$/, '');
+			return isIpPrivate(ip) ? 'local' : addr;
```

Comment:
I'd day let's stick with the "private" name.

Reply:
`local` seems to make more sense in this context, but for the sake of consistency - why not.